RESTEasy Reactive build is failing as `org.jboss.resteasy.spi.HttpRequest` is not available as bean

### Describe the bug

After #33267 we can't build our project. I didn't find anything about this change in migration guide nor in docs, so I think this is unexpected behavior.

### Expected behavior

Build do not fail.

### Actual behavior

```bash
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (build) on project http-advanced-reactive: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: Found 2 deployment problems: 
[ERROR] [1] Unsatisfied dependency for type org.jboss.resteasy.spi.HttpRequest and qualifiers [@Default]
[ERROR] 	- java member: org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider#request
[ERROR] 	- declared on CLASS bean [types=[jakarta.ws.rs.ext.MessageBodyReader<jakarta.ws.rs.core.Form>, org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider, java.lang.Object], qualifiers=[@Default, @Any], target=org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider]
[ERROR] [2] Unsatisfied dependency for type boolean and qualifiers [@Default]
[ERROR] 	- java member: org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider():useContainerParams
[ERROR] 	- declared on CLASS bean [types=[jakarta.ws.rs.ext.MessageBodyReader<jakarta.ws.rs.core.Form>, org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider, java.lang.Object], qualifiers=[@Default, @Any], target=org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider]
[ERROR] 	The following beans match by type, but none have matching qualifiers:
[ERROR] 		- Bean [class=java.lang.Boolean, qualifiers=[@ConfigProperty, @Any]]
[ERROR] 		- Bean [class=java.lang.Boolean, qualifiers=[@Any, @Claim("")]]
[ERROR] 	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1403)
[ERROR] 	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:305)
[ERROR] 	at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:156)
[ERROR] 	at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:526)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] 	Suppressed: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type org.jboss.resteasy.spi.HttpRequest and qualifiers [@Default]
[ERROR] 	- java member: org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider#request
[ERROR] 	- declared on CLASS bean [types=[jakarta.ws.rs.ext.MessageBodyReader<jakarta.ws.rs.core.Form>, org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider, java.lang.Object], qualifiers=[@Default, @Any], target=org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider]
[ERROR] 		at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:483)
[ERROR] 		at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:581)
[ERROR] 		at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:293)
[ERROR] 		... 13 more
[ERROR] 	Suppressed: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type boolean and qualifiers [@Default]
[ERROR] 	- java member: org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider():useContainerParams
[ERROR] 	- declared on CLASS bean [types=[jakarta.ws.rs.ext.MessageBodyReader<jakarta.ws.rs.core.Form>, org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider, java.lang.Object], qualifiers=[@Default, @Any], target=org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider]
[ERROR] 	The following beans match by type, but none have matching qualifiers:
[ERROR] 		- Bean [class=java.lang.Boolean, qualifiers=[@ConfigProperty, @Any]]
[ERROR] 		- Bean [class=java.lang.Boolean, qualifiers=[@Any, @Claim("")]]
[ERROR] 		at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:483)
[ERROR] 		at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:581)
[ERROR] 		at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:293)
[ERROR] 		... 13 more

```

### How to Reproduce?

Reproducer:

1. `git clone git@github.com:quarkus-qe/quarkus-test-suite.git`
2. `cd quarkus-test-suite/http/http-advanced-reactive`
3. `quarkus build`

### Output of `uname -a` or `ver`

Fedora

### Output of `java -version`

17

### GraalVM version (if different from Java)

22.3

### Quarkus version or git rev

999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

3.9.1

### Additional information

_No response_

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive-common/deployment/src/main/java/io/quarkus/resteasy/reactive/common/deployment/ResteasyReactiveCommonProcessor.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive-common/deployment/src/main/java/io/quarkus/resteasy/reactive/common/deployment/ResteasyReactiveCommonProcessor.java
@@ -46,6 +46,7 @@ import io.quarkus.arc.deployment.BeanArchiveIndexBuildItem;
 import io.quarkus.arc.deployment.BeanContainerBuildItem;
 import io.quarkus.arc.deployment.BuildTimeConditionBuildItem;
 import io.quarkus.arc.deployment.GeneratedBeanBuildItem;
+import io.quarkus.bootstrap.classloading.QuarkusClassLoader;
 import io.quarkus.deployment.Capabilities;
 import io.quarkus.deployment.Capability;
 import io.quarkus.deployment.annotations.BuildProducer;
@@ -79,7 +80,9 @@ public class ResteasyReactiveCommonProcessor {
     @BuildStep
     void searchForProviders(Capabilities capabilities,
             BuildProducer<AdditionalApplicationArchiveMarkerBuildItem> producer) {
-        if (capabilities.isPresent(Capability.RESTEASY) || capabilities.isPresent(Capability.REST_CLIENT)) {
+        if (capabilities.isPresent(Capability.RESTEASY) || capabilities.isPresent(Capability.REST_CLIENT)
+                || QuarkusClassLoader.isClassPresentAtRuntime(
+                        "org.jboss.resteasy.reactive.server.injection.JaxrsServerFormUrlEncodedProvider")) { // RESTEasy Classic could be imported via non-Quarkus dependencies
             // in this weird case we don't want the providers to be registered automatically as this would lead to multiple bean definitions
             return;
         }

